app/src/main/java/org/wikipedia/diff/ArticleEditDetailsFragment.kt

```diff
@@ -448,10 +575,87 @@ class ArticleEditDetailsFragment : Fragment(), WatchlistExpiryDialog.Callback, L
             .show()
     }
 
+    private fun showRollbackSnackbar() {
+        val snackbar = FeedbackUtil.makeSnackbar(requireActivity(), getString(R.string.patroller_tasks_patrol_edit_rollback_success))
+        snackbar.addCallback(object : Snackbar.Callback() {
+            override fun onDismissed(transientBottomBar: Snackbar, @DismissEvent event: Int) {
+                if (!isAdded) {
+                    return
+                }
+                showFeedbackOptionsDialog()
+            }
+        })
+        snackbar.show()
+    }
+
+    private fun showFeedbackOptionsDialog(skipPreference: Boolean = false) {
+        if (!skipPreference && !Prefs.showOneTimeRecentEditsFeedbackForm) {
+            return
+        }
+
+        var dialog: AlertDialog? = null
+        val feedbackView = layoutInflater.inflate(R.layout.dialog_patrol_edit_feedback_options, null)
+
+        val clickListener = View.OnClickListener {
+            viewModel.feedbackOption = (it as TextView).text.toString()
+            dialog?.dismiss()
+            if (viewModel.feedbackOption == getString(R.string.patroller_diff_feedback_dialog_option_satisfied)) {
+                // TODO: send to the event logging since it is satisfied
+                showFeedbackSnackbarAndTooltip()
+            } else {
+                showFeedbackInputDialog()
+            }
+        }
+
+        feedbackView.findViewById<TextView>(R.id.optionSatisfied).setOnClickListener(clickListener)
+        feedbackView.findViewById<TextView>(R.id.optionNeutral).setOnClickListener(clickListener)
+        feedbackView.findViewById<TextView>(R.id.optionUnsatisfied).setOnClickListener(clickListener)
+
+        dialog = MaterialAlertDialogBuilder(requireActivity())
+            .setTitle(R.string.patroller_diff_feedback_dialog_title)
+            .setCancelable(false)
```

Comment:
Do we really want to set it to `false` for cancellable? This forces the user to choose some option even when they accidentally choose it, hence tainting data. However, if it were to be the other way they can always choose it from menu when they accidentally cancel the dialog.